File: lib/snapshot.rb
#!/home/takayuk/local/bin/ruby -Ku
# encoding: utf-8
# coding: utf-8

require "json"
require "rexml/document"
require "open-uri"

def snapshot feed_url
  @response = REXML::Document.new(open(feed_url, :proxy=>"http://cache.st.ryukoku.ac.jp:8080"))

  @snapshot = Hash.new(0)
  @response.elements.each("rss/channel/item/") do |item|
    @snapshot.store(item.elements["link"].text, item.elements["description"].text)
  end

  @snapshot.freeze
end

if __FILE__ == $0
  snapshot(ARGV[0])
end

File: lib/snapshot.py
# encoding: utf-8
# coding: utf-8


def snapshot(url):

    import urllib
    from xml.dom import minidom, Node
    
    response = minidom.parse(urllib.urlopen(url))

    docids = []
    descs = []

    for entry in response.getElementsByTagName("item"):
        docids.append(entry.getElementsByTagName("link").item(0).childNodes[0].data)
        descs.append(entry.getElementsByTagName("description").item(0).childNodes[0].data)

    return zip(docids, descs)


if __name__ == "__main__":
    import sys
    args = sys.argv

    ss = snapshot(args[1])

    for s, t in ss:
        print(s)
        print(t)


